src/main/java/org/springframework/data/couchbase/repository/query/support/N1qlUtils.java

```diff
@@ -60,18 +63,15 @@
  * @author Mark Paluch
  */
 public class N1qlUtils {
+  //As per https://docs.couchbase.com/server/5.5/n1ql/n1ql-language-reference/indexing-meta-info.html
+  final static Set<String> META_PROPERTIES = new HashSet<>(Arrays.asList("id", "expiration", "cas"));
 
   /**
    * A converter that can be used to extract the {@link CouchbasePersistentProperty#getFieldName() fieldName},
    * eg. when one wants a path from {@link PersistentPropertyPath#toDotPath(Converter)} made of escaped field names.
    */
   public static final Converter<? super CouchbasePersistentProperty,String> FIELD_NAME_ESCAPED =
-      new Converter<CouchbasePersistentProperty, String>() {
-        @Override
-        public String convert(CouchbasePersistentProperty source) {
-          return "`" + source.getFieldName() + "`";
-        }
-      };
+   (source) -> META_PROPERTIES.contains(source.getFieldName()) ? "META()." + source.getFieldName() : "`" + source.getFieldName() + "`";
```

Comment:
What happens if the document has a normal property called "id", "cas", or "expiration" -- can the user still reference it?

Reply:
yes good point, will change the check to use the annotations instead and add the prefix.